contracts/Holdable.sol

```diff
@@ -0,0 +1,240 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "./IHoldable.sol";
+
+contract Holdable is IHoldable, ERC20 {
+
+   using SafeMath for uint256;
+
+    struct Hold {
+        address issuer;
+        address origin;
+        address target;
+        address notary;
+        bool expires;
+        uint256 expiration;
+        uint256 amount;
+        HoldStatusCode status;
+    }
+
+    mapping(string => Hold) private holds;
```

Comment:
To save strings on the blockchain is quite expensive. it is more efficient to convert them to byte32 and save that. We have done that in EM-Token for example, with a library: https://gitlab.com/iobuilders/em-token/blob/master/contracts/PayoutableToken.sol#L9